### PR TITLE
Track ingredient level

### LINKED ISSUE #10: Database function to retrieve last entry of certain type

### LINKED ISSUE #16: Track ingredient levels in database

### LINKED ISSUE #17: Ability to get status on ingredients by sending mail to JVM mail

### PR DESCRIPTION
**Pull request by MathiasMehl**
Added logic in check mail functions for mails with labels "menu, failures, clean"
Now the check every 5mins will get all unread mails, instead of just a fixed number. 
Tracks changes to ingredient level based on content in ingredientlevel, menu, and dispensedrinkevent mails. 
Made setup database function, assumes the coffee machine is filled when first starting (i.e was filled when the first event arrived, probably not true, but no way of getting the correct amount)
Can query and get the current ingredient levels by sending a mail with subject "ingredient-status" to  "fklubjvmloss@gmail.com"
Added new table just for tracking ingredient levels

This Resolves #16, Resolves #17, Resolves #10

### CODE REVIEW

File: mailclient.py [hunk 1]
```diff
@@ -20,11 +20,115 @@
...
+else:
+    JVM_MAIL = 'fklubjvmlosstest@gmail.com'
+
+MAX_COFFEE = 2400
+MAX_MILK = 1800
+MAX_SUGAR = 2400
+MAX_CACAO = 2200
+
+# Todo QA these dictionaries
+coffee_beans_usage = {"Café au lait": 8,
```
Comment (cogitantium) [inline mailclient.py]: Pull these out into a discrete file, maybe?

File: mailclient.py [hunk 2]
```diff
@@ -20,11 +20,115 @@
...
+               "Filter kaffe": 0,
+               "Filter koffie": 0,
+               "Filter coffee milk": 0,
+               "Filter coffee milk sugar": 4,
+               "Filter coffee sugar": 4,
+               "Hot water": 0,
+               "Latte macchiato": 0,
+               "Latte macchiato sugar": 4,
+               "Wiener melange": 4}
+
```
Comment (falkecarlsen) [inline mailclient.py]: Pull these out into a discrete file, maybe?

File: sqlite.py [hunk 3]
```diff
@@ -3,86 +3,131 @@
...
+
+def get_last_event_by_type(conn: sqlite3.Connection, type: str):
+    cur = conn.cursor()
+    select_last_query_by_type = f'SELECT * FROM {DISPENSE_TABLE_NAME} ' \
+                                f'WHERE "event_type"=? ORDER BY timestamp DESC LIMIT 1'
+    cur.execute(select_last_query_by_type, (type,))
+    return cur.fetchall()
+
+
+def get_last_event_by_type_older_than(conn: sqlite3.Connection, type: str, upper_time: int):
```
Comment (falkecarlsen) [inline sqlite.py:111]: Should have an inverse function as well.

**Review by falkecarlsen — CHANGES_REQUESTED** [review]
Small comments, otherwise nice!

File: sqlite.py (hunk 3 above)
Comment (MathiasMehl) [reply sqlite.py]: Function added in commit fde9d383289f262cd6dd454e4792851f4a3b0306

Comment (MathiasMehl) [reply sqlite.py:111]: Added in ffc1ae70725ea53dd132dfccbc369c1fc71ba2df

File: mailclient.py (hunk 2 above)
Comment (MathiasMehl) [reply mailclient.py]: Added a seperate file for this in 040fbc32cbd6182f1f1b4a7572812778e3afe455. Also added max values into this file

**Review by falkecarlsen — APPROVED** [review]
Nice :nerd_face:

**Pull request merged**